### PR TITLE
tls/x509utils: SubjectPublicKeyBytes() and SHA1/SHA224/SHA256 shortcuts

### PR DESCRIPTION
**Pull request by amery**
to index and compare public keys

## Summary by CodeRabbit

- **New Features**
	- Introduced a new function to compute the SHA256 hash of the Subject Public Key.
	- Enhanced functionality for handling Subject Public Keys with SHA1 and SHA224 hashing.
	- Added functionality to extract raw bytes of the Subject Public Key.

### CODE REVIEW

**Comment by amery** [comment]
> This pre-allocation might slightly improve performance, especially if this function is called frequently.

please explain the difference

**Comment by amery** [comment]
`SubjectPublicKeySHA256()` added

File: tls/x509utils/spki.go [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+// SubjectPublicKeySHA256 returns the SHA224 hash of the SubjectPublicKey
+// of a [crypto.PublicKey]
+func SubjectPublicKeySHA256(pub crypto.PublicKey) (hash [sha256.Size]byte, err error) {
+	b, err := SubjectPublicKeyBytes(pub)
+	if err != nil {
+		return hash, err
+	}
+
+	return sha256.Sum256(b), nil
+}
```
[the remark this answers is not in the document]
Comment (amery) [reply tls/x509utils/spki.go]: oops

**Comment by amery** [comment]
> `47-68`: **LGTM with a potential optimization.**

@coderabbitai won't both give me a preallocated zeroed struct?

**Comment by amery** [comment]
wow, deepsource really hates sha1

**Review by karasz — APPROVED** [review]
+1

**Comment by karasz** [comment]
> wow, deepsource really hates sha1

It is there, nobody is forced to use it, and anyway it is a hash, who cares, could be even md4

**Comment by amery** [comment]
`darvaza.org/x/tls@v0.2.2` tagged

**Pull request merged**